src/schema/request-query-schema.js

```diff
@@ -0,0 +1,17 @@
+const Joi = require('joi');
+const { geoservicesEnvelopeObjSchema, geoservicesEnvelopeArraySchema } = require('./geometry-schema');
+const requestQuerySchema = Joi.object().keys({
+    resultRecordCount: Joi.number().positive().integer(),
+    resultOffset: Joi.number().positive().integer(),
+    num: Joi.number().positive().integer(),
+    inSR: Joi.number().positive().integer(),
+    outSR: Joi.number().positive().integer(),
+    where: Joi.string(),
+    geometry: Joi.when('geometryType', {
+        is: 'esriGeometryEnvelope',
+        then: Joi.alternatives(geoservicesEnvelopeObjSchema, geoservicesEnvelopeArraySchema)
+    }),
+    geometryType: Joi.string().valid('esriGeometryEnvelope', 'esriGeometryPolygon')
+}).unknown(true);
```

Comment:
I think `unknown(true)` does the work on `optional` . Do you mean specific fields?

Reply:
No, I think `unknown` means the schema allows other keys (keys you haven't defined) to be present without throwing an error.  But you need to make sure that the schema doesn't throw errors if any of the keys you listed are missing, because they are optional for the query.

Reply:
As discussed, `unknown(true)` will not throw error if any of the keys listed are missing.